src/utils/swallow.js

```diff
@@ -0,0 +1,9 @@
+const swallow = (fail, fn) => (...args) => {
```

Comment:
Is there a reason to put the fail first instead of the function?

Reply:
Not really

Reply:
In case you are using a redux thunk, the fail handling function usually is shorter than the success, and it can make it easier to understand

Reply:
`const fetchFormCategories = () =>
  swallow(
    (error, dispatch) => dispatch(fetchFormCategoriesError(error)),
    async dispatch => {
      dispatch(fetchFormCategoriesRequest());
      const { data } = await formServices.fetchFormCategories();
      dispatch(fetchFormCategoriesSuccess(data));
    }
  );`

Reply:
`
const fetchFormCategories = () =>
  swallow(
    (error, dispatch) => dispatch(fetchFormCategoriesError(error)),
    async dispatch => {
      dispatch(fetchFormCategoriesRequest());
      const { data } = await formServices.fetchFormCategories();
      dispatch(fetchFormCategoriesSuccess(data));
    }
  );
`